src/server/list_family.cc

```diff
@@ -572,46 +606,77 @@ OpResult<uint32_t> OpRem(const OpArgs& op_args, string_view key, string_view ele
     return it_res.status();
 
   auto it = it_res->it;
-  quicklist* ql = GetQL(it->second);
-
-  int iter_direction = AL_START_HEAD;
-  long long index = 0;
-  if (count < 0) {
-    count = -count;
-    iter_direction = AL_START_TAIL;
-    index = -1;
-  }
-
-  quicklistIter qiter;
-  quicklistInitIterator(&qiter, ql, iter_direction, index);
-  quicklistEntry entry;
+  size_t len = 0;
   unsigned removed = 0;
+
   int64_t ival;
 
   // try parsing the element into an integer.
   int is_int = lpStringToInt64(elem.data(), elem.size(), &ival);
 
-  auto is_match = [&](const quicklistEntry& entry) {
-    if (is_int != (entry.value == nullptr))
-      return false;
+  if (it->second.Encoding() == kEncodingQL2) {
+    QList* ql = GetQLV2(it->second);
+    QList::Where where = QList::HEAD;
 
-    return is_int ? entry.longval == ival : ElemCompare(entry, elem);
-  };
+    if (count < 0) {
+      count = -count;
+      where = QList::TAIL;
+    }
+
+    auto it = ql->GetIterator(where);
+    auto is_match = [&](const QList::Entry& entry) {
+      if (is_int) {
+        return entry.is_int() && entry.ival() == ival;
+      }
+      return entry == elem;
```

Comment:
return (is_int && entry.is_int() && entry.ival() == ival) || entry == elem

Reply:
will fix in the subsequent pr